### PR TITLE
redux

### PR DESCRIPTION
**Pull request by gersonhiggins**
merging feature/redux into develop
Add redux components for addBook, removeBook and categories
check linters and fix

### CODE REVIEW

File: src/components/redux/store.js
Comment (joskalenda) [inline src/components/redux/store.js:1]: - [ ] Am not sure to understand what is happening here, what you described in your pull request summary is not present in your code, maybe you have forget to push your last changes but as an example, this file is empty which should not be possible if you implemented the redux actions, dear @gersonhiggins.
- Please check [this](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions_v2.md) project requirement once more and make sure you have implemented the all modifications

**Review by joskalenda — CHANGES_REQUESTED** [review]
| STATUS: REQUIRED CHANGES ♻️  |
|--------------------------------|
| Hi @gersonhiggins , |
|Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there! |
|  |

## Highlights

- [ ]... ✅ green linters

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/redux/store.js
Comment (gersonhiggins) [reply src/components/redux/store.js:1]: this projects its only about configure redux, but your checking out on other codes, there is a redux directory into the project

Comment (gersonhiggins) [reply src/components/redux/store.js:1]: src/components/redux

Comment (gersonhiggins) [reply src/components/redux/store.js:1]: o i can see changes didnt get pushed

File: src/components/redux/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+};
+
+export const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    filterCategories: (state, action) => {
+      const filterCat = state.categories.filter((book) => book.cat === action.payload);
+      filterCat.status = 'w';
+    },
```
Comment (Ibizugbe) [inline src/components/redux/categoriesSlice.js]: We are required in this milestone to create a reducer that checks for the status (`checkStatus`) and returns `under construction`😉 inside our categorySlice😉.

image

```suggestion
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
```
```

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @gersonhiggins 👋👋, 

tryagain


Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights✅✅

- [x] Linters are passing👍.
- [x] Gitflow is used👍.


 ## Required Changes ♻️

- [ ] Kindly ensure to make use of the `ducks` pattern for your folder structure as this is a requirement in this milestone😉.
image
When obeying this pattern, your redux folder shouldn't be under your components folder but under same hierarchy as your components folder as they should be both directly under the `src`  folder👍.


_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Tufahel — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

Hi @gersonhiggins  :wave: 
 
Well done on implementing the changes. 👍 
Your project is complete now!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 👏🏻



:heavy_check_mark: Correct GitFlow.
:heavy_check_mark: The app is working fine.


## Optional suggestions 🤷‍♂️

- [ ] ***[OPTIONAL]*** Kindly use the `capital` letter as the first letter of a commit message as it is professional.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.
- You can mail me at [Mail](mailto:tufahel96@gmail.com) or ask any question at Slack, `Name: Tufahel`

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**